pkg/azurestore/azureservice.go

```diff
@@ -295,14 +290,17 @@ func blockIDIntToBase64(blockID int) string {
 	return blockIDBinaryToBase64(binaryBlockID)
 }
 
-func blockIDBase64ToInt(blockID string) int {
+func blockIDBase64ToInt(blockID *string) int {
 	blockIDBase64ToBinary(blockID)
 	return int(binary.LittleEndian.Uint32(blockIDBase64ToBinary(blockID)))
 }
 
-func isAzureError(err error, code string) bool {
```

Comment:
Could we keep this function and inspect errors from Azure by looking into their service code instead of just checking their error message? Using the service code appears more robust in the long term to me.

Reply:
yes, sure.. i did that